Reporting/ad-extensions-reporter.js

```diff
@@ -0,0 +1,384 @@
+    605: 'ITEM_6_FINAL_MOBILE_URL',
+    700: 'ITEM_7_HEADER',
+    701: 'ITEM_7_DESCRIPTION',
+    702: 'ITEM_7_PRICE',
+    703: 'ITEM_7_UNIT',
+    704: 'ITEM_7_FINAL_URL',
+    705: 'ITEM_7_FINAL_MOBILE_URL',
+    800: 'ITEM_8_HEADER',
+    801: 'ITEM_8_DESCRIPTION',
+    802: 'ITEM_8_PRICE',
+    803: 'ITEM_8_UNIT',
+    804: 'ITEM_8_FINAL_URL',
+    805: 'ITEM_8_FINAL_MOBILE_URL'
+}
+
+var promotion = {
+    name: 'Promotion',
+    1: 'PROMOTION TARGET',
+    2: 'DISCOUNT MODIFIER',
+    3: 'PERCENT OFF',
+    4: 'MONEY AMOUNT OFF',
+    5: 'PROMOTION CODE',
+    6: 'ORDERS OVER AMOUNT',
+    7: 'PROMOTION START',
+    8: 'PROMOTION END',
+    9: 'OCCASION',
+    10: 'FINAL URLS',
+    11: 'FINAL MOBILE URLS',
+    12: 'TRACKING URL',
+    13: 'LANGUAGE',
+    14: 'FINAL URL SUFFIX'
+}
+
+var extensionTypeToPlaceholderTables = {
+    1: sitelink,
+    2: call,
+    3: app,
+    7: location,
+    17: callout,
+    24: structuredSnippet,
+    30: affiliateLocation,
+    31: message,
+    35: price,
+    38: promotion
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+// Functions
+
+function main() {
+    var spreadsheet = checkSpreadsheet(spreadsheetUrl, "Output Spreadsheet");
+    var sheets = spreadsheet.getSheets();
+
+    Logger.log("Preparing spreadsheet");
+    var allExtensionsSheet = spreadsheet.getSheetByName("All Extensions");
+    if (allExtensionsSheet === null) {
+        allExtensionsSheet = spreadsheet.insertSheet("All Extensions");
+    }
+    allExtensionsSheet.clear();
+
+    var extensionNames = Object.keys(extensionTypeToPlaceholderTables).map(
+        function (key) { return extensionTypeToPlaceholderTables[key].name }
+    )
+    for (var i = 0, n = sheets.length; i < n; i++) {
+        if (extensionNames.indexOf(sheets[i].getName()) !== -1) {
+            spreadsheet.deleteSheet(sheets[i]);
+        }
+    }
+
+    // Get report
+    Logger.log("Getting report");
+    if (typeof includeExtensionsWithZeroImpressions === 'undefined') {
+        Logger.log("Warning: options variable includeExtensionsWithZeroImpressions not found. " +
+            "Continuing, and by default including extensions that have had no impressions today."
+        );
+        impressionsCondition = '';
+    } else {
+        if (includeExtensionsWithZeroImpressions === true) {
+            var impressionsCondition = '';
+        } else if (includeExtensionsWithZeroImpressions === false) {
+            var impressionsCondition = 'WHERE Impressions > 0 ';
+        } else {
+            throw ("Value of options variable includeExtensionsWithZeroImpressions must be a boolean, either true or false. " +
+                "Currently, it is: " + includeExtensionsWithZeroImpressions)
+        }
+    }
+
+    var placeholderFeedItemReport = AdsApp.report(
+        'SELECT PlaceholderType, AttributeValues, DisapprovalShortNames ' +
+        'FROM PLACEHOLDER_FEED_ITEM_REPORT ' +
+        impressionsCondition +
```

Comment:
could even put it here
```suggestion
        getZeroImpressionsCondition(includeExtensionsWithZeroImpressions) +
```

Reply:
Very nice idea! I couldn't get it to work simply though, with having the check outside the function for whether `includeExtensionsWithZeroImpressions` is declared, so left this bit how it was.